Update: troubleshooting section in migration guide (#3918)
Kotlin/dokka: docs/topics/dokka-migration.md

* To enable build cache, follow instructions in the [Gradle build cache documentation](https://docs.gradle.org/current/userguide/build_cache.html#sec:build_cache_enable).
* To enable configuration cache, follow instructions in the [Gradle configuration cache documentation](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:usage:enable ).

> ## Troubleshoot your migration

Reviewer: Hmm 🤔 - we could have Dokka migration troubleshooting instead? - or Troubleshooting Dokka migration? 

Alternatively, Dokka migration troubleshooting tips?

Author: I'll leave it as just "Troubleshooting", what do you think?

> In large projects, Dokka can consume a significant amount of memory to generate documentation. This issue can be due to Gradle's memory allocation limits, 

Reviewer: ```suggestion
In large projects, Dokka can consume a significant amount of memory to generate documentation. This can happen due to Gradle's memory allocation limits, 
```

Reviewer: alternatively:
In large projects, Dokka can consume a significant amount of memory to generate documentation.
This can sometimes exceed Gradle’s memory limits, especially when processing large volumes of data.

and Dokka's processing of large volumes of data, among other factors.

> When Dokka generation runs out of memory, Gradle throws exceptions like `java.lang.OutOfMemoryError: Metaspace`.

Reviewer 2: ```suggestion
When Dokka generation runs out of memory the Gradle build will fail, and Gradle can throws exceptions like `java.lang.OutOfMemoryError: Metaspace`.
```

Minor suggestion, I think it reads a bit more naturally.

Reviewer 3: A small iteration 🙂 
```suggestion
When Dokka generation runs out of memory, the build will fail, and Gradle can throw exceptions like `java.lang.OutOfMemoryError: Metaspace`.
```

> Active efforts are underway to improve Dokka's performance, although Gradle limitations exist.

Reviewer: ```suggestion
Active efforts are underway to improve Dokka's performance, although some limitations stem from Gradle.
```

> As a workaround, when experiencing memory exceptions, you can try:

Reviewer: ```suggestion
If you encounter memory issues, try these workarounds:
```

* [Increasing heap space](#increase-heap-space)
* [Running Dokka in-process](#run-dokka-in-process)

### Increase heap space

> One alternative to resolve memory issues is to increase the amount of Java heap memory for the Dokka generator process. 

Reviewer 3: ```suggestion
One way to resolve memory issues is to increase the amount of Java heap memory for the Dokka generator process. 
```

In the `build.gradle.kts` file, adjust the 
following configuration option:

```Kotlin
dokka {

> // Dokka generates a new process managed by Gradle

Reviewer 3: ```suggestion
    // Dokka generates a new process managed by Gradle
```

>     workerIsolation = ProcessIsolation {

Reviewer 4: This doesn't look like a correct syntax:
<img width="615" alt="image" src="https://github.com/user-attachments/assets/62dd569d-44be-4cbd-b035-4c0fc9978890">

The correct one is:
```kotlin
dokka {
    dokkaGeneratorIsolation = ProcessIsolation {
        minHeapSize = "128m"
    }

//    or just:
//    ProcessIsolation {
//        minHeapSize = "128m"
//    }
}
```

Author: @adam-enko  Can you help me check and confirm how the code snippet should look like?

Reviewer 4: Also worth to fix in KDoc: https://github.com/Kotlin/dokka/blob/15986f67452872ac13956d37f7cf090a5c6cfc58/dokka-runners/dokka-gradle-plugin/src/main/kotlin/DokkaExtension.kt#L258
cc @adam-enko

Reviewer 2: Yes, good catch, it should be `dokkaGeneratorIsolation`

```suggestion
    dokkaGeneratorIsolation = ProcessIsolation {
```

https://github.com/Kotlin/dokka/pull/3921

        // Configures heap size

>         minHeapSize = "4g"

Reviewer 4: Shouldn't we suggest to increase `maxHeapSize` instead?

Author: @adam-enko  Same here: Can you help me check and confirm how the code snippet should look like?

Reviewer 2: Yes, it should be `maxHeapSize` not `minHeapSize`.

```suggestion
        maxHeapSize = "4g"
```

Reviewer 2: KDoc fixed in PR https://github.com/Kotlin/dokka/pull/3921

    }
}
```

> The example above shows a heap size of `4g`; you need to adjust and test until you find the best solution for your build.

Reviewer: is this 4GB? - I think we could clarify it here

Reviewer: for example, could we say ? 🤔

> If you have to increase the memory by an unreasonable amount, for example, significantly more than the memory required for Gradle,

Author: I think we should leave Gradle's case just as an example as it is not the only case. I came up with another based on your suggestion (:

[report an issue on Dokka's GitHub repository](https://kotl.in/dokka-issues).

> You have to apply this configuration to each subproject. It's recommended that you configure Dokka in a convention 
> plugin applied to all subprojects.
>
{style="note"}

> ### Run Dokka in-process

Reviewer: how about - Run Dokka in a Gradle process?
Although in-process might work as well - it's a setting I suppose 🤔

When both the Gradle build and Dokka generation require a lot of memory, they may run as separate processes, 
consuming significant memory on a single machine.

To optimize memory usage, you can run Dokka within the same Gradle process instead of as a separate process. This 
allows you to configure the memory for Gradle once instead of allocating it separately for each process.

To run Dokka within the same Gradle process, adjust the following configuration option in the `build.gradle.kts` file:

> ```Kotlin

Reviewer 2: ```suggestion
```kotlin
```

dokka {
    // Runs Dokka in the current Gradle process

>     workerIsolation = ClassLoaderIsolation()

Reviewer 4: Here we also need to replace `workerIsolation` with `dokkaGeneratorIsolation`
